torchtune/models/clip/_text_encoder.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import copy
+from typing import Optional
+
+import torch
+from torch import nn, Tensor
+
+from torchtune.modules.attention_utils import _MaskType
+
+
+class CLIPTextEncoder(nn.Module):
+    """
+    Text encoder for CLIP.
+
+    Args:
+        layer (nn.Module): A single encoder layer.
+        final_norm (nn.Module): Callable that applies normalization to the output of the encoder
+        vocab_size (int): size of the vocabulary, default 49408
+        max_seq_len (int): context size, default 77
+        embed_dim (int): embedding/model dimension size, default 768
+        num_layers (int): number of transformer layers, default 12
+    """
+
+    def __init__(
+        self,
+        *,
+        layer: nn.Module,
+        final_norm: nn.Module,
+        vocab_size: int = 49408,
```

Comment:
same question here - do we want to hardcode these defaults if they're tied to a specific version of CLIP?

Reply:
I don't think it's that important since this is a one off model definition. The main thing is to also define this same number in the builder.